### PR TITLE
auth lib

### PR DESCRIPTION
**Pull request by ericcater**

### CODE REVIEW

**Comment by ericcater** [comment]
e2e don't pass

**Comment by ericcater** [comment]
e2e pass

File: libs/auth/src/auth/guards/local.guard.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ExecutionContext, Injectable } from '@nestjs/common';
+import { GqlExecutionContext } from '@nestjs/graphql';
+import { AuthGuard } from '@nestjs/passport';
+
+@Injectable()
+export class LocalAuthGuard extends AuthGuard('local') {
```
Comment (yurisim) [inline libs/auth/src/auth/guards/local.guard.ts:7]: Can we give a general explanation of what these decorators do?

File: libs/types/ods/src/entity/user.entity.ts [hunk 2]
```diff
@@ -3,15 +3,15 @@ import { Role, User } from '.prisma/ods/client';
...
   firstName: string;
   lastName: string;
+  enabled: boolean;
 
   @Field(() => Role)
-  role: Role;
-  id: string;
-  email: string;
-  enabled: boolean;
+  role: Role | null;
```
Comment (yurisim) [inline libs/types/ods/src/entity/user.entity.ts:14]: DId you mean to make this null?

**Review by yurisim — COMMENTED** [review]
Ignore comments about "Do we need this?" I'll remove them in another branch

File: libs/types/ods/src/entity/user.entity.ts (hunk 2 above)
Comment (ericcater) [reply libs/types/ods/src/entity/user.entity.ts:14]: It either needs to be null or there needs to be a default value.

Comment (yurisim) [reply libs/types/ods/src/entity/user.entity.ts:14]: Why does it need to be null?

File: apps/ods-api/schema.prisma [hunk 3]
```diff
@@ -1,102 +1,102 @@
...
+  lastName     String
+  role         Role?
+  refreshToken RefreshToken?
+  orgs         Org[]
+}
+
+model RefreshToken {
+  id          String   @id @default(uuid())
+  user        User     @relation(fields: [userId], references: [id])
+  userId      String   @unique
```
Comment (yurisim) [inline apps/ods-api/schema.prisma:30]: This @unique here may have unintended consequences, as long as we are sure we never create a refreshtoken when one exists for a user already it's fine.

Comment (ericcater) [reply apps/ods-api/schema.prisma:30]: Running format adds it.

**Comment by yurisim** [comment]
Don't merge this until I get one last look at it! :)

**Pull request merged**